app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/TEIDataFragment.java

```diff
@@ -215,11 +260,24 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
             if (requestCode == REQ_DETAILS) {
                 activity.getPresenter().init();
             }
+            if (requestCode == FilterManager.OU_TREE) {
+                adapter.notifyDataSetChanged();
+            }
         }
     }
 
     @Override
-    public Consumer<List<Event>> setEvents() {
+    public Flowable<String> observeStageSelection(Program currentProgram, Enrollment currentEnrollment) {
+        if (adapter == null) {
+            adapter = new EventAdapter(presenter, currentProgram, currentEnrollment);
+            //TODO: Add header itemDecoration(requires ANDROAPP-656 to be merge)
```

Comment:
Could this be done now ?

Reply:
Not part of the feature. It is a nice to have extra. Should be done one the dashboard gets totally redesigned